### ISSUE TITLE
Optimizations for floating point arithmetic

### ISSUE THREAD

**Issue by MJoergen** [commanderx16/x16-rom#67]
I have a number of ideas for optimizing the floating point arithmetic routines in the BASIC code.

For now I've forked this project and added the file [xadd.s](https://github.com/MJoergen/x16-rom/blob/master/basic/xadd.s), which contains an optimized version of the floating point addition. Additionally, I've added the file [test/add.bas](https://github.com/MJoergen/x16-rom/blob/master/test/add.bas) to do some rudimentary testing.

Benchmarking is difficult, because the execution time depends on the operand values. One data-point is the simple loop "FOR I=1 TO 10000; NEXT I", which now executes in 109 jiffies, as opposed to 116 jiffies in the original ROM. This corresponds to reducing each addition by - on average - 120 clock cycles.

I have (untested) ideas for optimization multiplication too, but I'm unsure how to proceed from here.

**Comment by kktos** [comment]
Looks like a PR is in order ;)

**Comment by MJoergen** [comment]
One thing we need to remember is that the original ROM has an entry point within the floating point addition routine, namely faddc, that is called from another place in the ROM. I believe it is used when converting a literal constant to floating point. So this needs to be handled. And then I guess we should remove the old code.

**Comment by kktos** [comment]
Just replace the code with yours and we're set. I think @mist64 did what was needed to keep the fixed addresses fixed ;)

**Comment by MJoergen** [comment]
I've just issued a Pull Request. I welcome any feedback.

**Comment by MJoergen** [comment]
I've added a SQR implementation to my branch. It is more than five times faster than the original implementation.

The new implementation is more accurate too: During testing I noticed e.g. that the original ROM gave the value
SQR(1017) = 31.8904375
whereas the new implementation gives the value
SQR(1017) = 31.8904374
The correct value according to my calculator is close to 31.890437438.

Furthermore, the old ROM gave a non-zero value for SQR(9)-3, whereas the new implementation gives zero. In fact the new implementation gives exact integer results for all integer squares, up until SQR(139*139) which is not exactly 139.
